etcdserver/api/v3rpc/watch.go

```diff
@@ -17,17 +17,22 @@ package v3rpc
 import (
 	"io"
 
+	"github.com/coreos/etcd/etcdserver"
 	pb "github.com/coreos/etcd/etcdserver/etcdserverpb"
 	"github.com/coreos/etcd/storage"
 	"github.com/coreos/etcd/storage/storagepb"
 )
 
 type watchServer struct {
+	clusterID int64
+	memberID  int64
+	server    etcdserver.Server
+	raftTimer etcdserver.RaftTimer
 	watchable storage.Watchable
 }
 
-func NewWatchServer(w storage.Watchable) pb.WatchServer {
-	return &watchServer{w}
+func NewWatchServer(s *etcdserver.EtcdServer) pb.WatchServer {
+	return &watchServer{int64(s.Cluster().ID()), int64(s.ID()), s, s, s.Watchable()}
```

Comment:
let's construct the watcherServer with named Fields. It improves readability when we have a few fields to fill in, especially some of them are filled in with same variable but for different interfaces.

```go
&watchServer